examples/data_pipeline/roberta_dataframe.py

```diff
@@ -0,0 +1,81 @@
+from argparse import ArgumentParser
+
+import torcharrow as ta
+import torcharrow.dtypes as dt
+import torcharrow.pytorch as tap
+import torchtext.transforms as T
+from torch.hub import load_state_dict_from_url
+from torch.nn import Module
+from torch.utils.data import DataLoader
+from torchtext.datasets import SST2
+
+
+class RobertaTransform(Module):
+    def __init__(self) -> None:
+        super().__init__()
+        # Instantiate various transforms
+
+        # Tokenizer to split input text into tokens
+        encoder_json_path = "https://download.pytorch.org/models/text/gpt2_bpe_encoder.json"
+        vocab_bpe_path = "https://download.pytorch.org/models/text/gpt2_bpe_vocab.bpe"
+        self.tokenizer = T.GPT2BPETokenizer(encoder_json_path, vocab_bpe_path)
+
+        # vocabulary converting tokens to IDs
+        vocab_path = "https://download.pytorch.org/models/text/roberta.vocab.pt"
+        self.vocab = T.VocabTransform(load_state_dict_from_url(vocab_path))
+
+        # Add BOS token to the beginning of sentence
+        self.add_bos = T.AddToken(token=0, begin=True)
+
+        # Add EOS token to the end of sentence
+        self.add_eos = T.AddToken(token=2, begin=False)
+
+    def forward(self, input: ta.DataFrame) -> ta.DataFrame:
+        input["tokens"] = input["text"].map(self.tokenizer.forward, dtype=dt.List(dt.string))
+        input["tokens"] = input["tokens"].list.slice(stop=254)
+        input["tokens"] = input["tokens"].map(self.vocab, dtype=dt.List(dt.int32))
+        input["tokens"] = input["tokens"].map(self.add_bos)
+        input["tokens"] = input["tokens"].map(self.add_eos)
+        return input
+
+
+def main(args):
+
+    # Instantiate transform
+    transform = RobertaTransform()
+
+    # Create SST2 datapipe and apply pre-processing
+    train_dp = SST2(split="train")
+
+    # convert to DataFrame of size batches
+    # TODO: Figure out how to create DataFrame of larger size and create batches consequently
```

Comment:
I'm trying to understand the benefit of doing this in the future. Assuming we can convert a larger chunk of the rows in the datapipe into a dataframe (based on system capabilities), why would we operate on a smaller batch rather than operating on the entire dataframe?

Reply:
So this is purely my lack of knowledge in terms of APIs. It was just out of convenience that I created DataFrame of size equal to batch-size such that to_tensor can yield the batched tensor (since to_tensor converts the whole DataFrame into tensor). Ideally we would like to create larger DataFrames, and apply batch transformations, followed by creation of batched tensors of size batch-size. Perhaps we can use batch API. Let me explore a bit on this.  @wenleix wondering if you have any suggestions here?

Reply:
I guess my question here was why would this followup be necessary rather than maintaining the status quo of what you implemented here? Why would we want to create a larger DataFrame rather than always keeping DataFrame size equal to batch size?

Reply:
hmm good question. I guess, we should do the benchmarking to figure this out. I don't know the internal dynamics, but what if creating one large dataframe of size `nxbatch_size` takes time less than `n` times the time taken to create single DataFrame of size `batch_size`? Also it might be more efficient to run transform on larger batch size instead of making `n` calls? But it could be that it just turns out to be same in which case this is already an elegant API use :).

Reply:
So if batch size is small (e.g. 16 rows), the dispatch and other Python framework overhead in eager mode might be non-negligible. That being said, we haven't benchmarked and have an understanding about "what's small".

In general we should have a wrapper (e.g. `StreamingDataFrame` @VitalyFedyunin had prototyped  ) that separates the  "in memory buffer" size (say 4096) with the "batch size" (say 16 or 128), so we don't need to let user to worry about such.

Reply:
> In general we should have a wrapper (e.g. StreamingDataFrame @VitalyFedyunin had prototyped ) that separates the "in memory buffer" size (say 4096) with the "batch size" (say 16 or 128)

This sounds interesting!